wordpress-mobile/WordPress-Android
# Prepublishing : Old draft state

Fixes #12197

Shared to get thoughts. 

## Findings
When a draft is opened in the Post List or Editor and the date is in the past it gets cleared.
## Solution
`PublishPostImmediatelyUseCase` was used to add this functionality to the Editor & Post List.

## Testing
1. Go to the Post List. 
2. Press Publish on an old draft. 
3. Notice it says Immediately instead of the old date. 
[Video Example](https://drive.google.com/file/d/1n7GQCYQqMP5-DwtXmV8vO_ev4RyLLDkn/view?usp=sharing)
--------

## Testing
1. Go to the Post List. 
2. Edit an old draft.
2. Press Publish (if you change the status to Draft, the outcome below works). 
3. Notice it says Immediately instead of the old date. 

[Video Example](https://drive.google.com/file/d/12XZorKwwQSvrar6RL-Qig4yurZa0A-fG/view?usp=sharing)

So the above happens because when a post is not new it's deemed as `Published` so this logic needs to be improved.

## Reviewing 
Only 1 reviewer is needed but anyone can review. 

## Submitter Checklist
